source/vir/src/well_formed.rs

```diff
@@ -372,6 +373,17 @@ fn check_one_expr(
                 VisitorControlFlow::Stop(e) => Err(e),
             }?;
         }
+        ExprX::AssertBy { ensure, vars, .. } => match &ensure.x {
+            ExprX::Binary(crate::ast::BinaryOp::Implies, _, _) => {
+                if !vars.is_empty() {
+                    diags.push(VirErrAs::Warning(
+                        error(&expr.span, "using ==> in `assert forall` will not assume the antecedent in the body, and is deprecated, consier using `implies` instead of `==>`
```

Comment:
I think we want `; consider` instead of `, consier`.

Thinking about it more, I worry that this warning might confuse people (or perhaps I'm just confused).  To me, the warning sounds like we want everyone to stop using `==>` in assertions about `forall`.  But that's not actually what we want long-term; in the near future, we want them to only use `==>` and not `implies`.

Perhaps instead we could say:
```
using ==> in `assert forall` does not currently assume the antecedent in the body.  If you want to assume the antecedent, consider using `implies` instead of `==>`.  If you do not intend to assume the antecedent, please ...
```

Reply:
Thanks for the input. I wasn't particularly happy with my formulation, and I do like your formulation better. I'll make the change. @jaybosamiya-ms @jaylorch any other opinions on this?

Reply:
I agree that we shouldn't say that it's deprecated, because it's going to be the way to go soon.

Reply:
How about something like:
> using `==>` in `assert forall` does not currently assume the antecedent in the body.  This behavior is intended to be changed.  If you want to assume the antecedent, consider using `implies` instead of `==>`.  If you do not intend to assume the antecedent, then please report to the Verus developers with your specific use case (with `--record`), to aid the Verus developers in making an informed decision.

It _is_ quite verbose, but it is intended to be a short term warning, that goes away when we deprecate `implies`, so I'd rather it carry enough information for people to report it to us, than it be missed.